plugins/workspace-plugin/src/ca-cert.ts

```diff
@@ -0,0 +1,45 @@
+/**********************************************************************
+ * Copyright (c) 2019-2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+const PUBLIC_CRT_PATH = '/public-certs';
+const SS_CRT_PATH = '/tmp/che/secret/ca.crt';
+
+const CA_BUNDLE_PATH = '/tmp/ca-bundle.crt';
+
+export const getCertificate = new Promise<string | undefined>(async resolve => {
```

Comment:
I believe it would be better to have some shared code with a certificate-related functionality
instead of copying it the third time, in addition to [this](https://github.com/eclipse/che-theia/blob/e165f775c878f87a124d1ecf1b2bdad5c1ed4bc9/plugins/task-plugin/src/machine/websocket.ts#L197-L214) and [this](https://github.com/eclipse/che-theia/blob/e165f775c878f87a124d1ecf1b2bdad5c1ed4bc9/extensions/eclipse-che-theia-remote-api/src/node/che-server-certificate-service-impl.ts#L21-L41).

Reply:
it could be provided by `@eclipse-che/plugin` then ? (so all plug-ins can benefits from it ? )

Reply:
I think it's not a big deal to copy twenty lines of code.
But need to think about your proposal. Let's leave it as improvement.
In comparing with places of code you posted, workspace plugin creates CA bundle in `/tmp`.

Reply:
@benoitf well, trying to add it to `@eclipse-che/plugin`

Reply:
on a more global problem, I'm not sure why we're using curl instead of Axios

we could just provide an axios object pre-configured with correct certificates from che/plug-in ext

Reply:
Whatever the solution, I would prefer this be merged sooner rather than later because it's a blocker for CRW. We can always iterate on the fine tuned items later.